dbms/src/Storages/StorageDistributed.cpp

```diff
@@ -167,6 +167,9 @@ StorageDistributed::StorageDistributed(
         if (num_local_shards && remote_database == database_name && remote_table == table_name)
             throw Exception("Distributed table " + table_name + " looks at itself", ErrorCodes::INFINITE_LOOP);
     }
+
+    /// HACK: disable metadata for StorageDistributed queries
+    const_cast<Context &>(context).getSettingsRef().insert_sample_with_metadata = false;
```

Comment:
Why it's hack? :)

Won't it disable setting forever?

Reply:
It's hack cause of class interface that says context should be const here.
It should disable the setting only for this query. It's a query context too.